src/KernelAbstractions.jl

```diff
@@ -303,7 +303,7 @@ abstract type Device end
 abstract type GPU <: Device end
 
 struct CPU <: Device end
-struct CUDA <: GPU end
+struct CUDAGPU <: GPU end
```

Comment:
I am not a fan of the name, maybe we rename it to `NVIDIA` like the prospective names below?

But I don't have a super strong opinion on this.

Reply:
yeah I have no issues changing it, I'll switch to the consensus.

NVIDIA wouldn't clash with CUDA and aligns with the other names so it seems like a good candidate.

Reply:
I was going to use `CUDABackend` but I'm not sure that is better as it's a little long

Reply:
What names are you thinking about for the AMD and Intel GPU backends?

Reply:
it seems `AMD` and `Intel`?

Reply:
There are a lot of ` device isa CUDA` in the CLIMA code, so would `CUDADevice` be a better name?

Reply:
Either way sounds okay with me.